addons/binding/org.openhab.binding.tankerkoenig/src/main/java/org/openhab/binding/tankerkoenig/internal/utility/Holiday.java

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.tankerkoenig.internal.utility;
+
+import org.joda.time.DateTime;
+import org.joda.time.LocalDate;
+
+/**
+ * Utility static class to calculate if a given date is a german holiday.
+ * Presently only those holidays that are valid in whole germany (Bundesweit) are used!
+ * Code copied from Openhab1-Addons Samples-Rules
+ *
+ * @author Jürgen Baginski
+ *
+ */
+public final class Holiday {
+
+    private Holiday() {
+        // avoid instantiation.
+    }
+
+    public static boolean isHoliday(DateTime now) {
+        // Checks if today is a German holiday (Feiertag im ganzen Bundesgebiet!)
+        // Code from Openhab1-Addons Samples-Rules
+        int year = now.getYear();
+        int a = year % 19;
+        int b = year / 100;
+        int c = year % 100;
+        int d = b / 4;
+        int e = b % 4;
+        int f = (b + 8) / 25;
+        int g = (b - f + 1) / 3;
+        int h = (19 * a + b - d - g + 15) % 30;
+        int i = c / 4;
+        int k = c % 4;
+        int l = (32 + 2 * e + 2 * i - h - k) % 7;
+        int m = (a + 11 * h + 22 * l) / 451;
+        int month = (h + l - 7 * m + 114) / 31;
+        int day = ((h + l - 7 * m + 114) % 31) + 1;
+        boolean holiday = false;
+        LocalDate easterSunday = LocalDate.parse(year + "-" + month + "-" + day);
+        LocalDate stAdvent = LocalDate.parse(year + "-12-25")
+                .minusDays(((LocalDate.parse(year + "-12-25").getDayOfWeek()) + 21));
+        int dayOfYear = now.getDayOfYear();
+        // bundesweiter Feiertag "Neujahr"
+        if (dayOfYear == LocalDate.parse(year + "-01-01").getDayOfYear()) {
+            holiday = true;
+        }
+        // Baden-Württemberg, Bayern, Sachsen-Anhalt, Heilige 3 Könige
+        else if (dayOfYear == LocalDate.parse(year + "-01-06").getDayOfYear()) {
+            holiday = false;
```

Comment:
Yes, there are a lot of holidays that are not relevant for the whole country. I'd like to keep them for later usage, when the option to include them is in cooperated. I'm presently not sure how to make the selection, either on PaperUI (that much On/Off????) or in config file.  Another way could be to use a calender.